### ISSUE TITLE
Kraken margin trades not recognized / Tax report completely wrong

### ISSUE THREAD

**Issue by XL-Reaper** [rotki/rotki#1979]
Hi,

I've connected Rotki 1.10.1 to my Kraken API and tried to create a tax report.
The report is just completely wrong and the issue may be that according to the report I have 0 "Margin Positions Profit Loss" and just everything is in "General Trade Profit Loss".

There are lots of margin trades on my account for the timeframe I tried the tax report for, but it seems they are either not exported by the API (no idea how to check that) or that Rotki is matching them to a normal trade instead of a margin trade.

Did I miss something in the setup for this to work correctly?

Please let me know how I can help to debug this issue.

### System Description

Operating system: Windows 10 2004
Rotki version: 1.10.1

**Comment by LefterisJP** [comment]
Hey @XL-Reaper!

We do not support margin trades in any centralized exchange yet. This is why you don't see any margin trades in the tax report.

To follow the implementation of margin trade support, follow this issue: https://github.com/rotki/rotki/issues/1980

Closing this as it's not a bug but a feature request that already exists.

**Comment by uvok** [comment]
With Rotki v1.22.1, I connected to the Kraken Exchange with a fresh API key (all "query..." permissions). Under History > Trades, all trades appear. (Same trades as in Kraken web interface).

However, I did some margin trades, and Rotki seems to consider them as regular trades, leading to a wrong P&L calculation. (as mentioned by issue opener)

I know that margin trades are not supported yet (#1980). However, I think margin trades should be ignored instead for the time being? But I guess it will still be handled under #1980 ?

**Comment by LefterisJP** [comment]
Hey @uvok. Margin trades should not appear under normal trades, but to be honest I can't say since I don't know what the kraken api returns for margin trades. Their docs don't help too much.

I would need to test with an api key that has done some margin trades. If you can share on such read only key that would help us.

**Comment by uvok** [comment]
I would rather not disclose the API key (or at least share it only privately), however, I tried to do some debugging myself.

When I call [rotkehlchen/exchanges/kraken.py, line 636, `self.query_until_finished`](https://github.com/rotki/rotki/blob/ecb603444c693c8bc864aa4dda116554c66aa6b1/rotkehlchen/exchanges/kraken.py#L636) myself with the full range, it appears there's a "margin" key.

```
{'cost': '40.37320',
  'fee': '0.11304',
  'margin': '20.18660',
  'misc': '',
  'ordertxid': 'OOP6QG-XXXXX-XXXXX',
  'ordertype': 'limit',
  'pair': 'DOTEUR',
  'posstatus': 'closed',
  'postxid': 'TKH2SE-XXXXX-XXXXX',
  'price': '40.37320',
  'time': 1636742401.8571,
  'type': 'buy',
  'vol': '1.00000000'},
 {'cost': '299.75514',
  'fee': '0.77936',
  'margin': '0.00000',
  'misc': '',
  'ordertxid': 'OGZ45Z-XXXXX-XXXXX',
  'ordertype': 'limit',
  'pair': 'XETHZEUR',
  'postxid': 'TKH2SE-XXXXX-XXXXX',
  'price': '4076.67000',
  'time': 1636741375.303,
  'type': 'buy',
  'vol': '0.07352941'},
```

Also, trying to look up a margin trade as shown on the website:

```
# order position
>>> [r for r in result if r["ordertxid"]=="O4OC23-Q7XYE-23FL5L"]
[{'ordertxid': 'O4OC23-Q7XYE-23FL5L', 'postxid': 'TKH2SE-M7IF5-CFI7LT', 'posstatus': 'closed', 'pair': 'XXBTZEUR', 'time': 1637489943.6994, 'type': 'buy', 'ordertype': 'market', 'price': '52358.90000', 'cost': '199.99948', 'fee': '0.54000', 'vol': '0.00381978', 'margin': '39.99990', 'misc': ''}]
# opening trade id as per website
>>> [r for r in result if r["postxid"]=="TNTOSL-4HULG-ECNOOL"]
[{'ordertxid': 'OVXWLZ-IDGOH-EPXPR5', 'postxid': 'TNTOSL-4HULG-ECNOOL', 'pair': 'XXBTZEUR', 'time': 1637493651.5927, 'type': 'sell', 'ordertype': 'stop market', 'price': '52130.30000', 'cost': '199.12628', 'fee': '0.51773', 'vol': '0.00381978', 'margin': '39.82526', 'misc': 'closing'}]
# closing trade id - not found for whatever reason
>>> [r for r in result if r["postxid"]=="T3XE3Y-MDAKF-7OUBNA"]
[]
# order id as shown in closing trade details
>>> [r for r in result if r["ordertxid"]=="OVXWLZ-IDGOH-EPXPR5"]
[{'ordertxid': 'OVXWLZ-IDGOH-EPXPR5', 'postxid': 'TNTOSL-4HULG-ECNOOL', 'pair': 'XXBTZEUR', 'time': 1637493651.5927, 'type': 'sell', 'ordertype': 'stop market', 'price': '52130.30000', 'cost': '199.12628', 'fee': '0.51773', 'vol': '0.00381978', 'margin': '39.82526', 'misc': 'closing'}]
```

For regular trades, the 'margin' key is 0. Does that help?

**Comment by LefterisJP** [comment]
> For regular trades, the 'margin' key is '0.00000' (not sure if it also has this exact amount of precision) Does that help?

Hey @uvok yes that helps. We can indeed skip trades that have non zero margin for now, as it seems they are bundled with the rest in the history of trades. This must have changed in their api since when we first wrote it only spot trades were returned.

>  would rather not disclose the API key (or at least share it only privately), however, I tried to do some debugging myself.

Understand. And definitely would never ask you to share api key publically. Most people who do that to help us with developing new exchange connectors or to debug problems DM me or any other team member in discord: https://discord.com/invite/aGCxHG7

Email also works. support@rotki.com
